### DISCUSSION TITLE
درباره ایرانسل

### DISCUSSION THREAD

**Discussion by GFW-knocker** [GFW-knocker/gfw_resist_tls_proxy#93] [Q&A]
هر چیزی درباره فیتلرینگ ایرانسل میدونید به اشتراک بگذارید
با همکاری همدیگه ایرانسل رو پاره خواهیم کرد

**Comment by GFW-knocker** [comment]
روی ایرانسل
چه ابزارهایی کار میکنه؟
چه پروتکل هایی کار میکنه؟
دقیق توضیح بدید مثلا سرور شخصی ، کلادفلر ، شدوساکس فلان ، ایپی تمیز فلان ، ...

**Reply by Arya2010** [reply]
روی ایرانسل الان آتلاین کار میکنه . سرعتشم بد نیست 
سرورها VPS آمازون یا گوگل کلاد هستن ( چون دیجیتال اوشن کار نمیکنه ) 
ما الان فعلا تجربه مثبتمون با شدوساکس آتلاین هست .

**Reply by TKZ700** [reply]
اوتلاین(شدوساکس)روش خوب کار میکنه
reality هم روی سرور شخصی همینطور
vps ها هم AWS

**Comment by GFW-knocker** [comment]
یه چیزی که عجیبه :
ما یه ip کلادفلر داریم 162.159.134.42  که سایت plos.org هم پشتش هست
حالا رو این ایپی سایت plos.org بعد دو ثانیه تاخیر با سرعت باز میشه
ولی سایت پشت کلودفلر ما مثلا ali.site خیلی خیلی کند هست
خوب میایم sni رو 100 تیکه میکنیم ولی باز کند هست
سوال اینه:
ایپی که یکسانه. دامنه رو هم که نمیبینه. از کجا میفهمه ما چی تبادل میکنیم؟
ایا در مسیر برگشت در قسمت server hello  ، اطلاعاتی وجود داره که مشخص بشه؟
چت gpt میگه از server hello اسم سرور رو درمیاره 
راست میگه فکر کنم
اونوقت ما باید یک cert fake به کلودفلر بدیم  با اون وصل شیم. شدنیه یعنی؟

**Reply by GFW-knocker** [reply]
دمت گرم. عالی.
داریم راهش رو پیدا میکنیم
لینک هم خواستید پاک کنید

**Reply by ghost** [reply]
اگه خواستید این Unbound رو هم مطالعه بکنید. شاید بتونه مشکل Latency بالای DOH رو حل کنه. این ویدیوهایی که هستن بیشتر برا Local Network هستن و نمیدونم برا اینترنت هم همون قابلیت رو داشته باشه یا نه. طبق این ویدیو دفعه اول که داخل Local Network پینگ میگیره، latency بالاس ولی از دفعات بعدی کاملا اوکی میشه.

[YouTube 1](https://youtu.be/XbbziN_H71U)
[YouTube 2](https://youtu.be/DOJ39lyx6Js)
[Unbound - GitHub](https://github.com/NLnetLabs/unbound)

**Comment by Arya2010** [comment]
دیشب متوجه شدیم که آتلاین هفته پیش یک آپدین داشته که من کدش رو اینجا میزارم 
@@ -93,18 +93,13 @@ func findEntry(firstBytes []byte, ciphers []*list.Element) (*CipherEntry, *list.
	chunkLenBuf := [2]byte{}
	for ci, elt := range ciphers {
		entry := elt.Value.(*CipherEntry)
		id, cipher := entry.ID, entry.CryptoKey
		saltsize := cipher.SaltSize()
		salt := firstBytes[:saltsize]
		cipherTextLength := 2 + cipher.TagSize()
		cipherText := firstBytes[saltsize : saltsize+cipherTextLength]
		_, err := shadowsocks.DecryptOnce(cipher, salt, chunkLenBuf[:0], cipherText)
		cryptoKey := entry.CryptoKey
		_, err := shadowsocks.Unpack(chunkLenBuf[:0], firstBytes[:cryptoKey.SaltSize()+2+cryptoKey.TagSize()], cryptoKey)
		if err != nil {
			debugTCP(id, "Failed to decrypt length: %v", err)
			debugTCP(entry.ID, "Failed to decrypt length: %v", err)
			continue
		}
		debugTCP(id, "Found cipher at index %d", ci)
		// Move the active cipher to the front, so that the search is quicker next time.
		debugTCP(entry.ID, "Found cipher at index %d", ci)
		return entry, elt
	}
	return nil, nil

کاری که این کد داره میکنه اینه که به لیستی ار سایفرها مراجعه میکنه و خودش گزینه مناسب رو انتخاب میکنه . 
حدسمون اینه که ممکنه این جایگزین پوشاننده SNI باشه که داریم روش کار میکنیم . 
این کد رو اینجا به اشتراک گذاشتم شاید بقیه بتونن کمک کن که ببینیم مشکل چیه ؟ توی سایفره یا توی finger print؟ 

اینم لینک کد کامل آپدیت شده اس که به زبون Go برای شدوساکس نوشته شده : 
https://github.com/Jigsaw-Code/outline-ss-server/commit/7798d19e350454a4ecd1f7727a403aa149b49785

**Reply by GFW-knocker** [reply]
مرسی خیلی لطف کردید
برا اتلاین دیگه اسم دامنه و اینا نمیدید؟
مثلا مثل reality بگید دامنه مایروسافت باشه و اینا؟
یا تنظیم خاصی داشته باشه؟

**Reply by GFW-knocker** [reply]
تا اینجا حدسم اینه که ایرانسل پکت های برگشت رو رصد میکنه بجا رفت

**Reply by Arya2010** [reply]
عطف به سوال دامنه برای آتلاین  : الان به آتلاین دامنه و اینا نمیدیم و بدون دامنه داره روی ایرانسل کار میکنه . دلیلشو نمیدونیم اتفاقا خیلی دنبال این هستم که دلیلیشو در بیارم . اگه از دوستان کسی میدونه خوشحال میشم به اشتراک بزارین . 
دیشب هم مجددا متوجه شدم که یک بازنویسی مجدد در عرض ۲۴ ساعت گذشته توی کدهای آتلاین بوده چرا که خیلی ها گزارش عدم وصل آتلاین روی همراه اول رو داده بودن 
شاید خوب باشه شما هم یه نگاهی به کدها بندازین 
حالا در رابطه با نکته دوم : اینکه حدس اینه که ایرانسل پکتهای برگشت رو رصد میکنه سوالم اینه که آیا نمیشه از TLS 1.3 استفاده کرد که چه توی Client Hello  و چه توی Server Hello  همه اطلاعات رو کد میکنه ؟ منجمله SNI  رو ؟ 
نظر شما چیه ؟

**Reply by GFW-knocker** [reply]
ممنون از پاسخ شما
خیر ،TLS 1.3 همین الان در حال استفاده است. اون نسخه Encrypted Client Hello ECH هنوز در مرحله draft هست . هنوز پشتیبانی درستی ازش نمیشه حتی کلادفلر اسما میگه پشتیبانی میکنیم ولی عملا کار نمیکنه و اینکه مبتنی بر DNS هست که همین الان فیلتره.

**Reply by tsalehm** [reply]
> تا اینجا حدسم اینه که ایرانسل پکت های برگشت رو رصد میکنه بجا رفت

احتمالا میتونه چنتا حالت باشه هر کدوم ممکنه و نه. من دانش فنی خیلی زیادی ندارم پس حدسیات خودم رو میگم. اینکه اول از همه ممکنه فرگمنت ها رو سر هم کرده باشه یا حداقل بخشی از اونا رو سر هم کرده باشه و ببینه همون یزره بخش مث سرور های عادی نیست. 
اگه تاخیر بین فرگمنت ها و تیکه هاشون مقدار متقارن و برابری داشته باشن بنظر تشخیص نسبتا راحت تریه با همون 3 تا فرگمنت اولی شاید یجوری این تشخیص و انجام میده. 
پس بنظرم بدک نیست فرگمنت ها به بخش های نامتقارن و در زمان های نامتقارن تقسیم و ارسال بشن. 

یکم بعیده بنظر من که فیلترینگ بر اساس سرور هلو باشه چون سرور هلو اطلاعات در مورد سرور ارائه نمیده


و اینکه هنوزم این احتمال وجود داره که کلا ایرانسل پهنای باند رو برای دامنه هایی که توی وایت لیست مقدسش نیستن رو پایین اورده. این مشکل با سی دی ان اروان هم مشاهده میشه در صورتی که دیدم با ایپی تمیز یا تانل خوب کار می کنه که احتمال اینکه ایرانسل داره یه sni وایت لیست اجرا می کنه رو بیشتر می کنه.

**Reply by tsalehm** [reply]
البته یه نکته دیگه هم هست حتما از tls1.3 استفاده بشه چون tls1.2 توی سرور هلو یکسری اطلاعات اضافی هست با اونها احتمال زیاد بشه تشخیص داد که این ویژگی توی tls 1.3 رمزگزاری میشه
اینم گفتم احتمال اینکه ایرانسل داره محدودیت پهنای باند ایجاد می کنه خیلی زیاده من خودم روی سرورم چند بار چه با اروان چه با کلاد و... خواستم یه فایل رو دانلود کنم با یه سرعت محدود انجام میشد درصورتی که روی یکسری سایت ها با همون سی دی ان سرعت 100 برابر هست 

خب ایرانسل شاید میگه کانکشنی که نمیتونم سرهم کنم رو بیام با پهنای باند محدود ادامه بدم!

**Reply by GFW-knocker** [reply]
> خب ایرانسل شاید میگه کانکشنی که نمیتونم سرهم کنم رو بیام با پهنای باند محدود ادامه بدم!

دقیقا داره همین کارو میکنه تو منطقه ما. همه چیز محدوده.  هیچ پکتی رد نمیشه مگر مشخص باشه. و با این کار رسما نت خارجش قطعه. حتی دانلود از مایکروسافت قطعه. حتی سایت های مجاز دو سه ثانیه مکث داره تا سرعت بالا بره. اینو دیگه فقط با کوکتل میشه باز کرد.
